### PR TITLE
PORTALS-646

### PR DESCRIPTION
**Pull request by leem42**
[PORTALS-646](https://sagebionetworks.jira.com/browse/PORTALS-646) - added kebab menu

Also:
* Put table component into its own folder
* Broke apart table component into smaller components
* Replaced modals in table with bootstrap modals which eliminates additional overhead of tracking each modal's state, now they function on their own.

### CODE REVIEW

File: src/lib/containers/table/table-top/ColumnSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const {
+    headers,
+    isColumnSelected,
+    visibleColumnCount = Infinity,
+    toggleColumnSelection
+  } = props
+  return (
+    <Dropdown.Menu alignRight={true}>
+      {
+        headers.slice(0,17).map(
```
Comment (conundrumgirl) [inline src/lib/containers/table/table-top/ColumnSelection.tsx]: would it make sense to declare 17 as a const with a descriptive name?

File: src/demo/containers/Demo.tsx [hunk 2]
```diff
@@ -126,26 +126,27 @@ class Demo extends React.Component<{}, DemoState> {
...
     token = ''
     return (
       <div>
-        <p className="App-intro text-center">Synapse production version: {this.state.version}</p>
-        {
-          (this.state.token && this.state.token !== '') &&
+        <p className="App-intro text-center">
+          Synapse production version: {this.state.version}
+        </p>
+        {this.state.token && this.state.token !== '' && (
```
Comment (conundrumgirl) [inline src/demo/containers/Demo.tsx]: is `this.state.token !== ''` needed?  I think `this.state.token` will be falsy if this.state.token == ''

File: src/__tests__/lib/containers/SynapseTable.test.tsx [hunk 3]
```diff
@@ -89,69 +88,73 @@ describe('basic functionality', () => {
...
-      // Double check its not showing be default
+    it('isModalDownloadOpen opens the ModalDownload', async () => {
+      const { wrapper, instance } = await createShallowComponent(props)
+      // Verify its not showing by default
       expect(wrapper.find(ModalDownload)).toHaveLength(0)
-      // Click dropdown open
-      await wrapper.find(`.${DOWNLOAD_OPTIONS_CONTAINER_CLASS} button`).at(0).simulate('click')
-      // Click the dropdown
-      await wrapper.find(`.${DOWNLOAD_OPTIONS_CONTAINER_CLASS} li`).at(0).simulate('click')
+      await instance.toggleStateVariables('isModalDownloadOpen')(stubbedEvent)
```
Comment (leem42) [inline src/__tests__/lib/containers/SynapseTable.test.tsx]: get rid of!

File: src/lib/containers/table/SynapseTable.tsx [hunk 4]
```diff
@@ -0,0 +1,1062 @@
...
+    // open this in a new window on synapse.org
+    window.open(
+      `https://www.synapse.org/#!Synapse:${parsed.synId}/tables/query/${encodedQuery}`,
+      '_self',
+    )
+  }
+
+  private renderDropdownDownloadOptions = () => {
+    const { isExpanded } = this.state
+    const toggleStateArgs: BooleanKeys<SynapseTableState>[] = [
```
Comment (leem42) [inline src/lib/containers/table/SynapseTable.tsx]: get rid of BooleanKeys<SynapseTableState>[]

Comment (leem42) [reply src/lib/containers/table/SynapseTable.tsx]: prefer passing arguments explicitly

File: src/lib/containers/table/SynapseTable.tsx [hunk 5]
```diff
@@ -0,0 +1,1062 @@
...
+    )
+  }
+
+  private renderTableTop = (headers: SelectColumn[]) => {
+    const { title } = this.props
+    const { isExpanded } = this.state
+    const tooltipAdvancedSearchId = 'openAdvancedSearch'
+    const { colorPalette } = getColorPallette(this.props.rgbIndex!, 1)
+    const background = colorPalette[0]
+    const onDownloadTableOnlyArguments: BooleanKeys<
```
Comment (leem42) [inline src/lib/containers/table/SynapseTable.tsx]: change this

**Pull request merged**